File: Python/caesar.py
# cs50 pset 6
from cs50 import get_string
import sys


def main():
    if len(sys.argv) != 2:
        print("Usage: ./caesar key") # Remind user to giv an argument

    # converts the word into int
    key = int(sys.argv[1])

    word = get_string("Plaintext: ")
    print("Cyphertext: ", end="")

    #loop through each char in str and add the k ints to it
    fix = 65 if word.isupper() else 97
    for i in range(len(word)):

        # convert strings to unicode numbers
        a = ord(word[i])
        b = ((((a - fix) + key) % 26) + fix)
        c = chr(b)
        print(c, end="")

    print()

if __name__ == "__main__":
    main()

File: C/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <ctype.h>

// main takes two argumaents: an int and a string
int main(int argc, string argv[])
{
    if (argc != 2)
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
    int key = atoi(argv[1]) % 26; // if argv[1] > 26, keep the remainder instead
    string word = get_string("Plaintext:");

    for(int i = 0; i < strlen(word); i++)
    {

        // all of this weird nonsense is required to loop back around to a if the letters go past z
        int fix = isupper(word[i]) ? 65 : 97;
        int pi = word[i] - fix;
        int ci = (pi + key) % 26;
        printf("%c", ci + fix);

    }
    printf("\n");
    return 0;
}

